RTP Header Processing

fluffy opened the pull request.

juberti commented on draft-uberti-avtcore-cryptex.md:
```diff
@@ -45,6 +45,22 @@ Signaling
+
+## Receiving
+
+After decrypting and authenticating the packet, if there is an RPT
```
Here I would just say that 0xC0DE extensions are processed exactly like the one-byte headers defined in S 4.2 of 8285. We also need to define a magic tag for two-byte headers  (see S 4.3 of 8285, perhaps, 0xC02, or 0x200)

murillo128 replied:
what do we do with the appbits of the 2 bytes header extension? I think we should just not allow to use them with cryptex.

juberti commented on draft-uberti-avtcore-cryptex.md:
```diff
@@ -45,6 +45,22 @@ Signaling
 
+## Sending
+
+When sending an RTP packet that requires any header extensions to a
```
As noted below, we should point people at 8285 and simply say what is different than the normal processing.

murillo128 commented on draft-uberti-avtcore-cryptex.md:
```diff
@@ -54,6 +70,8 @@ Security Considerations
 IANA Considerations
 ===================
 
+TODO: allocate RTP header extension code 0xC0DE to this spec
```
Is this really needed? I have failed to find where the 0xBEDE or 0x100 values are registered in IANA for 1 bytes or 2 bytes extension.

fluffy replied:
I will remove this

murillo128 commented on draft-uberti-avtcore-cryptex.md:
```diff
@@ -45,6 +45,22 @@ Signaling
+
+After decrypting and authenticating the packet, if there is an RPT
+header extension with a 16 bit RTP header extension tag of 0xC0DE, it
+MUST be changed from 0xC0DE to 0xBEDE so that it can be processed as a
```
I think this in an implementation detail, we should only specify the format that goes over the wire. If the implementation takes the normal profile id as input and changes it when doing the encryption to the encrypted id (and viceversa on decryption) or not is something we should not cover in the spec.

ibc replied:
Honestly I think this should be done by the srtp encrypt/decrypt call. I see zero reasons to change the content of the extensions block without also replacing the proper id value that indicates how such a content is.

ibc replied:
This is like if you mangle a packet by decrypting its payload without changing the encrypted bit/flag/id. No sense IMHO. Just update it so the new packet represents full valid information.

JonathanLennox replied:
My suggestion would be to add an informational note, that implementations may wish to perform the reverse transformation on decryption.  I agree we should only be specifying the wire format, but this is an obvious implementation technique.

ibc replied:
Ok, sorry, I understood that it was the header extension id reverse transformation should be done by the application (e.g. the RTP endpoint). Let me fix my concern: it should be done at SRTP stack level when invoking the encrypt/decrypt calls on a packet.

fluffy replied:
Agree with Inaki should be done in libSRTP before the app gets the packet. I'll try and make that clearer. What i was really trying to say is after you inverse the encryption transform, this is processed just like normal header extension.

fluffy commented:
due to the way the rebase showed up, I'm just going to close and start a new PR for this but I think I addressed the comments so will be good to have a look  at new one
